### PR TITLE
feat: add token command line options

### LINKED ISSUE #30: Add command line option for access tokens
`--gh-token` and `--npm-token`

See #19

### PR DESCRIPTION
**Pull request by jo**
Add two command line options for authentication tokens:
`--gh-token`: Github personal access token
`--npm-token`: NPM access token

Closes #30

### CODE REVIEW

**Comment by gr2m** [comment]
Looks great to me, thanks @jo

File: src/lib/github.js [hunk 1]
```diff
@@ -80,41 +81,57 @@ function createAuthorization (info, cb) {
...
-    answers.password = answers.password || passwordStorage.get(answers.username)
 
-    info.github = answers
-    info.github.endpoint = info.ghepurl || 'https://api.github.com'
-
-    createAuthorization(info, (err, data) => {
-      if (err) {
-        log.error('Could not login to GitHub. Check your credentials.')
-        return cb(err)
+    if ('gh-token' in info.options) {
```
Comment (christophwitzko) [inline src/lib/github.js]: Can we use `_.has` here? :speak_no_evil: 
https://lodash.com/docs#has

Comment (christophwitzko) [inline src/lib/github.js]: `if (_.has(info.options, 'gh-token')) {`

File: src/lib/npm.js [hunk 2]
```diff
@@ -49,27 +49,40 @@ module.exports = function (pkg, info, cb) {
       name: 'username',
       message: 'What is your npm username?',
       default: conf.get('username'),
-      validate: _.ary(_.bind(validator.isLength, null, _, 1), 1)
+      validate: _.ary(_.bind(validator.isLength, null, _, 1), 1),
+      when: function (answers) {
+        return !('npm-token' in info.options)
```
Comment (christophwitzko) [inline src/lib/npm.js]: https://github.com/semantic-release/cli/pull/31/files#r40338596

File: src/lib/npm.js [hunk 3]
```diff
@@ -49,27 +49,40 @@ module.exports = function (pkg, info, cb) {
...
+      }
     }, {
       type: 'input',
       name: 'email',
       message: 'What is your npm email?',
       default: conf.get('email'),
-      validate: validator.isEmail
+      validate: validator.isEmail,
+      when: function (answers) {
+        return !('npm-token' in info.options)
```
Comment (christophwitzko) [inline src/lib/npm.js]: https://github.com/semantic-release/cli/pull/31/files#r40338596

File: src/lib/npm.js [hunk 4]
```diff
@@ -49,27 +49,40 @@ module.exports = function (pkg, info, cb) {
...
+      }
     }, {
       type: 'password',
       name: 'password',
       message: 'What is your npm password?',
       validate: _.ary(_.bind(validator.isLength, null, _, 1), 1),
       when: function (answers) {
+        if ('npm-token' in info.options) return false
```
Comment (christophwitzko) [inline src/lib/npm.js]: https://github.com/semantic-release/cli/pull/31/files#r40338596

File: src/lib/npm.js [hunk 5]
```diff
@@ -49,27 +49,40 @@ module.exports = function (pkg, info, cb) {
...
         if (!info.options.keychain) return true
         if (info.options['ask-for-passwords']) return true
         return !passwordStorage.get(answers.username)
       }
     }], (answers) => {
-      answers.password = answers.password || passwordStorage.get(answers.username)
       info.npm = answers
 
+      if ('npm-token' in info.options) {
```
Comment (christophwitzko) [inline src/lib/npm.js]: https://github.com/semantic-release/cli/pull/31/files#r40338596

**Comment by christophwitzko** [comment]
LGTM :+1:

File: src/lib/github.js (hunk 1 above)
Comment (jo) [inline src/lib/github.js]: aha, lodash fan here :smile:

sure, will do that change now...

Comment (christophwitzko) [inline src/lib/github.js]: @jo thank you! :relaxed:

**Comment by jo** [comment]
@christophwitzko I applied the changes you proposed and squash-force-pushed the commits here. Could you have a look at it again, please?

Thanks :100:

**Comment by christophwitzko** [comment]
:ok_hand: :+1:

**Pull request merged**